src/test-app/__tests__/expectBoundFunction.test.ts

```diff
@@ -0,0 +1,14 @@
+import { expectBoundFunction } from '../../lib';
+import { setCount } from '../actions';
+import { userReducer } from '../reducers';
+import { increaseCounter } from '../sagas/user-sagas';
+
+test('Test helper with mocked sleep call', () => {
+  return expectBoundFunction(increaseCounter)
+    .withReducer(userReducer)
+    .toDispatch(setCount({ count: 1 }))
+    .call()
```

Comment:
`whenCalledWith`?

Also wondering whether this should be moved closer to the `expectBoundFunction`. What do you think, can this be phrased more fluidly?

Reply:
`whenCalledWith` sounds better. We could move it after the `withMocks` call which would kind of follow the actual flow of the function a little bit more.

Reply:
Do you think we should change this also in `expectSaga`?

Reply:
Yes, I think for `expectSaga` it would also improve readability if we move it up. 

Not sure about `whenDispatched` since that reads like the saga would be dispatched. 

Maybe `whenStartedWith`?

Reply:
Sound good, let's do the change in a different PR though